app/code/Magento/Ui/Controller/Adminhtml/Index/Render.php

```diff
@@ -30,6 +30,7 @@ public function execute()
         $component = $this->factory->create($this->_request->getParam('namespace'));
         $this->prepareComponent($component);
         $this->_response->appendBody((string) $component->render());
+        $this->_response->setHeader('Content-Type', 'application/json', true);
```

Comment:
Good point, I wasn't sure if all responses were in json format (didn't take my time to look it through, my mistake).
I'll have a look at a better solution.